doc/user/content/overview/arrangements.md

```diff
@@ -0,0 +1,130 @@
+
+When creating an arrangement for a join where the user has not specified a key, Materialize attempts to choose a key that will ensure that data is well distributed. If there is a primary key, that will be used; if there are source fields not required by the query, they are not included. Often Materialize can pull primary key info from a Confluent schema.
+
+### Arrangement size
+
+For joins, the size of an arrangement, or amount of memory it requires, is roughly equivalent to its number of distinct `(data, time)` pairs, which can be small even if the number of records is large. As an illustration, consider a histogram of taxi rides grouped by the number of riders and the fare amount. The number of `(rider, fare)` pairs will be much smaller than the number of total rides that take place.
+
+The amount of memory that the arrangement requires is then further reduced by background [compaction](/ops/deployment/#compaction) of historical data.
+
+### Example arrangements
+
+Let's take a look at some of the arrangements that Materialize creates for different circumstances.
+
+#### `COUNT`
+
+Some of the simplest arrangements are those for Differential Dataflow operators. For example, the `COUNT` operator, which Materialize uses for internal logging, has two arrangements:
+
+* An arrangement for the input, which indexes the key whose values will be counted
+* An arrangement for the output, which maintains the results
+
+`COUNT` reads from the input arrangement to count values, and both reads from the output (to see the previous total) and writes to the output (adding the new updates to the previous total).
+
+![Diagram of arrangements for count](/images/arrangements-count.png "Diagram of arrangements for COUNT")
+
+#### Three-way join
+
+Let's say that you want to review orders by shipdate. The query itself is:
+
+```sql
+SELECT
+    l_orderkey,
+    o_orderdate,
+    o_shippriority,
+    sum(l_extendedprice * (1 - l_discount)) AS revenue
+FROM
+    customer,
+    orders,
+    lineitem
+GROUP BY
+    l_orderkey,
+    o_orderdate,
+    o_shippriority
+ORDER BY
+    revenue desc,
+    o_orderdate;
+```
+
+The query is a three-way join between `customer`, `orders`, and `lineitem`, followed by a reduction. For the purposes of this explanation, we're going to focus on the join and skip the reduction.
+
+When joining the three tables, Materialize builds indexes on `customers` + `orders`  and (`customers` + `orders`) + `lineitem`. It can do this because it separates out the work of the join calculation from the work of indexing information or creating arrangements. Each of the original columns and both of the necessary joins create arrangements that can be reused for other queries.
```

Comment:
Maybe a nit, but Materialize builds arrangements, not indexes. We have a bit of a problem that there is a clash of terms going on here.

Nit: but pluralization is `customer`, `orders`, `lineitem`. The problem is that `order` is a reserved keyword in SQL, and the example would/should have used that except that it cannot.

I'm not clear on the notation  `customers` + `orders`  and (`customers` + `orders`) + `lineitem`. It seems like some of the "+" things mean "comma" and one of them means "after joining these two". I would say that MZ builds arrangements on `customer`, `orders`, `lineitem`, and the result of joining `customer` with `orders`. 

I think this example is probably not the easiest. What do you think about starting with a binary join: two inputs, which need to be arranged by the columns that will be equated. Then building up a three-way join, which is like the binary join, except that in addition to arranging the first two inputs, MZ now has to arrange an intermediate relationship that the user might not have known about.